### DISCUSSION TITLE
Make it possible to specify the saver irrespective of the output file name

### DISCUSSION THREAD

**Discussion by aradalvand** [libvips/libvips#3579] [Q&A]
Similar to #1348
I have a scenario where I'm using various vips command-line tools and commands (e.g. `vipsthumbnail`, `vips arrayjoin`, etc.) and I need to save the output as an extensionless file; but currently the only way you can specify the intended output format is by specifying an extension.

It would be nice if we could specify the format/saver separately, like so:
```
vipsthumbnail input.jpg --format webp --output image_file
```

This is possible in ImageMagick, for example.

**Comment by jcupitt** [comment]
Hi @aradalvand,

You can always just call the saver directly, would that work for you?

```
$ vips webpsave x.jpg banana
```

Writes a WebP file called `banana`.

**Comment by aradalvand** [comment]
Hi!
But how do would you use `webpsave` with `vipsthumbnail` for example?

**Comment by jcupitt** [comment]
You can thumbnail to stdout, so perhaps

```
$ vipsthumbnail k2.jpg -o .webp > banana
```

**Comment by jcupitt** [comment]
That works for all vips commands, eg.:

```
vips invert k2.jpg .ppm | vips rot stdin .webp d90 > banana
```

Chains two vips commands together, with a PPM intermediate.

It's much better to use eg. python, of course, since that gets you proper chaining.
